src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/executor/TableSchemaObject.java

```diff
@@ -1,22 +1,107 @@
+import com.datastax.oss.driver.api.core.data.ByteUtils;
+import com.datastax.oss.driver.api.core.metadata.schema.ColumnMetadata;
+import com.datastax.oss.driver.api.core.metadata.schema.IndexMetadata;
 import com.datastax.oss.driver.api.core.metadata.schema.TableMetadata;
+import com.datastax.oss.driver.api.core.type.VectorType;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.stargate.sgv2.jsonapi.exception.SchemaException;
+import io.stargate.sgv2.jsonapi.service.schema.SimilarityFunction;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
 
 public class TableSchemaObject extends TableBasedSchemaObject {
 
   public static final SchemaObjectType TYPE = SchemaObjectType.TABLE;
 
-  public TableSchemaObject(TableMetadata tableMetadata) {
+  private final VectorConfig vectorConfig;
+
+  private TableSchemaObject(TableMetadata tableMetadata, VectorConfig vectorConfig) {
     super(TYPE, tableMetadata);
+    this.vectorConfig = vectorConfig;
   }
 
   @Override
   public VectorConfig vectorConfig() {
-    return VectorConfig.notEnabledVectorConfig();
+    return vectorConfig;
   }
 
   @Override
   public IndexUsage newIndexUsage() {
     return IndexUsage.NO_OP;
   }
+
+  /**
+   * Get table schema object from table metadata
+   *
+   * @param tableMetadata
+   * @param objectMapper
+   * @return
+   */
+  public static TableSchemaObject getTableSettings(
+      TableMetadata tableMetadata, ObjectMapper objectMapper) {
+    Map<String, String> extensions =
+        (Map<String, String>)
+            tableMetadata.getOptions().get(CqlIdentifier.fromInternal("extensions"));
+    String vectorize = extensions != null ? extensions.get("vectorize") : null;
+    Map<String, VectorConfig.ColumnVectorDefinition.VectorizeConfig> resultMap = new HashMap<>();
+    if (vectorize != null) {
+      try {
+        String vectorizeJson =
+            new String(ByteUtils.fromHexString(vectorize).array(), StandardCharsets.UTF_8);
+        // Convert JSON string to Map
+        JsonNode vectorizeByColumns = objectMapper.readTree(vectorizeJson);
+        Map<String, VectorConfig.ColumnVectorDefinition.VectorizeConfig> vectorizeConfigMap =
+            new HashMap<>();
+        while (vectorizeByColumns.fields().hasNext()) {
+          Map.Entry<String, JsonNode> entry = vectorizeByColumns.fields().next();
+          VectorConfig.ColumnVectorDefinition.VectorizeConfig vectorizeConfig =
+              objectMapper.treeToValue(
+                  entry.getValue(), VectorConfig.ColumnVectorDefinition.VectorizeConfig.class);
+          vectorizeConfigMap.put(entry.getKey(), vectorizeConfig);
+        }
+      } catch (JsonProcessingException | IllegalArgumentException e) {
```

Comment:
Ok, one last request: was hoping for separate catch blocks -- so for `readTree()` generic "Bad JSON" (because that'd be causing it) -- and then for `treeToValue()` can indicate specific vector field that was bad.